src/v2/spec.yaml

```diff
@@ -3996,6 +4000,72 @@ components:
           $ref: "#/components/schemas/Signer"
         user:
           $ref: "#/components/schemas/User"
+    RegisterUserOnChainReqBody:
+      type: object
+      required:
+        - registration
+      properties:
+        registration:
+          type: object
+          required:
+            - deadline
+            - signature
+            - custody_address
+            - recovery_address
+          properties:
+            deadline:
+              type: integer
+              minimum: 0
+              example: 1715190000
+            signature:
+              $ref: "#/components/schemas/0xHexSignature"
+            custody_address:
+              $ref: "#/components/schemas/EthAddress"
+            recovery_address:
+              $ref: "#/components/schemas/EthAddress"
+        storage_units:
+          type: integer
+          minimum: 0
+          example: 2
+        signers:
+          type: array
+          items:
+            type: object
+            required:
+              - signature
+              - metadata
+              - public_key
+              - deadline
+            properties:
+              signature:
+                $ref: "#/components/schemas/0xHexSignature"
+              metadata:
+                $ref: "#/components/schemas/0xHexSignature"
+              public_key:
+                $ref: "#/components/schemas/Ed25519PublicKey"
+              key_type:
+                type: integer
+                minimum: 1
+                default: 1
+              metadata_type:
+                type: integer
+                minimum: 1
+                default: 1
+              deadline:
+                type: integer
+                minimum: 0
+                example: 1715190000
+    RegisterUserOnChainResponse:
+      type: object
+      properties:
+        message:
+          type: string
+          example: Account registered successfully
+        transaction_hash:
+          type: string
+          example: 0xabc123...
```

Comment:
@Shreyaschorge add an actual transaction hash that is pointing to account registration